dtbase/models/arima.py

```diff
@@ -0,0 +1,550 @@
+import logging
+import os
+import sys
+from copy import deepcopy
+from datetime import timedelta
+from typing import Optional, Tuple, Union
+
+import coloredlogs
+import numpy as np
+import pandas as pd
+import pydantic
+from sklearn.metrics import mean_absolute_percentage_error, mean_squared_error
+from sklearn.model_selection import TimeSeriesSplit
+from statsmodels.tsa.statespace.sarimax import SARIMAX, SARIMAXResultsWrapper
+
+from dtbase.models.utils.config import read_config
+from dtbase.models.utils.dataprocessor.clean_data import clean_data, clean_data_list
+from dtbase.models.utils.dataprocessor.config import (
+    ConfigData,
+    ConfigOthers,
+    ConfigSensors,
+)
+from dtbase.models.utils.dataprocessor.get_data import get_training_data
+from dtbase.models.utils.dataprocessor.prepare_data import prepare_data
+from dtbase.services.base import BaseModel
+
+# Read default params from config file
+_config_file_path = os.path.join(
+    os.path.dirname(os.path.realpath(__file__)), "config_arima.ini"
+)
+_defaults = read_config(_config_file_path, "arima")
+
+# set up logging
+logger = logging.getLogger(__name__)
+logging.basicConfig(stream=sys.stdout, level=logging.INFO)
+field_styles = coloredlogs.DEFAULT_FIELD_STYLES
+
+# change the default levelname color from black to yellow
+field_styles["levelname"]["color"] = "yellow"
+coloredlogs.ColoredFormatter(field_styles=field_styles)
+coloredlogs.install(level="INFO")
+
+
+class ConfigArima(pydantic.BaseModel):
+    hours_forecast: int = pydantic.Field(
+        default=_defaults["hours_forecast"], validate_default=True
+    )
+    arima_order: tuple[int, int, int] = pydantic.Field(
+        default=_defaults["arima_order"], validate_default=True
+    )
+    seasonal_order: tuple[int, int, int, int] = pydantic.Field(
+        default=_defaults["seasonal_order"], validate_default=True
+    )
+    trend: str | list[int] | None = pydantic.Field(
+        default=_defaults["trend"], validate_default=True
+    )
+    alpha: float = pydantic.Field(default=_defaults["alpha"], validate_default=True)
+    perform_cv: bool = pydantic.Field(
+        default=_defaults["perform_cv"], validate_default=True
+    )
+    cv_refit: bool = pydantic.Field(
+        default=_defaults["cv_refit"], validate_default=True
+    )
+
+
+class ArimaModel(BaseModel):
+    def __init__(self, config: Optional[dict] = None) -> None:
+        super().__init__()
+        self.config = config
+
+    @staticmethod
+    def _report_errors_for_data(data: pd.Series) -> None:
+        """
+        Report errors if the input data is not a pandas Series indexed by timestamp.
+        """
+        if not isinstance(data.index, pd.DatetimeIndex):
+            logger.error(
+                "The time series on which to train the ARIMA model must be indexed by "
+                "timestamp."
+            )
+            raise ValueError
+
+    def get_config_from_config_files(self) -> ConfigArima:
```

Comment:
Could this be renamed to something like `set_config_defaults`? It only reads values from the files for configuration variables that are not yet set in `self.config`.